modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -90,20 +89,22 @@ const reduceInlineTextElements = (editor: Editor, merge: boolean | undefined): v
 
     Tools.each(dom.select('*[data-mce-fragment]'), (node) => {
       const isInline = Type.isNonNullable(textInlineElements[node.nodeName.toLowerCase()]);
-      if (isInline && StyleUtils.hasInheritableStyles(dom, node)) {
-        for (let parentNode = node.parentElement; Type.isNonNullable(parentNode) && parentNode !== root; parentNode = parentNode.parentElement) {
-          // Check if the parent has a style conflict that would prevent the child node from being safely removed,
-          // even if a exact node match could be found further up the tree
-          const styleConflict = StyleUtils.hasStyleConflict(dom, node, parentNode);
-          if (styleConflict) {
-            break;
+      if (isInline) {
+        // Remove nodes recursively if identical to any parent node
```

Comment:
Could this comment be removed if you renamed the function to `unwrapAllEqualChildElements` or something that captures what it's actually doing.